Add score normalization and combination documentation (#4985)
opensearch-project/documentation-website: _query-dsl/compound/hybrid.md

---
layout: default
title: Hybrid
parent: Compound queries
grand_parent: Query DSL
nav_order: 70
---

# Hybrid query

This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, see the associated [GitHub issue](https://github.com/opensearch-project/neural-search/issues/244).    

> {: .warning}

Reviewer: This is not an experimental feature now. We have fixed the flag.

Reviewer 2: Feature is enabled by default as Navneet mentioned, it's possible to disable it using a feature flag ```plugins.neural_search.hybrid_search_disabled``` (corresponding PR  https://github.com/opensearch-project/neural-search/pull/274)

> Use a hybrid query to combine relevance scores from multiple queries into one score for a given document. A hybrid query contains a list of one or more queries, which are run in parallel at the data node level. It calculates document scores at the shard level independently for each subquery. The subquery rewriting is done at the coordinating node level to avoid duplicate computations.

Reviewer: ```suggestion
Use a hybrid query to combine relevance scores from multiple queries into one score for a given document. A hybrid query contains a list of one or more queries and calculates document scores at the shard level independently for each subquery. The subquery rewriting is done at the coordinating node level to avoid duplicate computations.
```

Author: The blog is mentioned in the normalization processor

> ## Example

Reviewer: can we have a full example of how to do hybrid search here. Example including the creation of Search Pipeline and then the hybrid query

Author: Added a link to the example in the normalization processor documentation so it can be maintained in one place.
